ext/zip/php_zip.c

```diff
@@ -1257,6 +1257,7 @@ PHP_FUNCTION(zip_read)
 
 		zr_rsrc->zf = zip_fopen_index(rsrc_int->za, rsrc_int->index_current, 0);
 		if (zr_rsrc->zf) {
+			Z_ADDREF_P(zip_dp);
```

Comment:
This *might* lead to a memory leak. Wouldn't the refcount increase for every `zip_read()`?

Reply:
ASAN nor DEBUG build reported anything on this new test nor in zip_read.phpt

Reply:
Argh – resources! For a clean solution, the `le_zip_entry` resources would need to store a reference to their `le_zip_dir` resource; otherwise, as happens here, if the `le_zip_dir` resource is closed, there are dangling pointers. Increasing the refcount here (and possibly elsewhere) only shades the problem. Given that the procedural API is deprecated as of PHP 8.0.0, I'd opt for wontfix. @remicollet, what do you think?

Reply:
If that s obsolete for that long, might be best remove it for 8.5/9.0 or so many people still using this api ?

Reply:
I think the plan is to remove for PHP 9 (removing in a minor is quite a BC break). And I guess most are using the OO-API already.